boards/nuvoton/numaker_m2l31ki/numaker_m2l31ki.dts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <nuvoton/m2l31kid.dtsi>
+#include "numaker_m2l31ki-pinctrl.dtsi"
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	model = "Nuvoton NuMaker M2L31KI board";
+	compatible = "nuvoton,numaker-m2l31ki";
+
+	aliases {
+		led0 = &red_led;
+	};
+
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		red_led: led_0 {
+			gpios = <&gpioc 14 GPIO_ACTIVE_LOW>;
+			label = "User LD1";
+		};
+	};
+};
+
+&scc {
+	/* For USB 1.1 Host/Device/OTG, configure to 192MHz, which can generate necessary 48MHz. */
+	/* For USB 2.0 Host/Device/OTG or no USB application, comment out to use default. */
+	core-clock = <DT_FREQ_M(192)>;
+};
+
+&gpioc {
+	status = "okay";
+};
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x00008000>;
+		};
+		slot0_partition: partition@8000 {
+			label = "image-0";
+			reg = <0x00008000 0x00038000>;
+		};
+		slot1_partition: partition@40000 {
+			label = "image-1";
+			reg = <0x00040000 0x0003e000>;
+		};
```

Comment:
This will not work for MCUboot, the sizes depend upon what operating mode you are going to use:

- swap using move: images can be same size but this is inefficient, slot0 should be larger than slot1, the most efficient size can be determined by building e.g. `samples/subsys/mgmt/mcumgr/smp_svr` for your board using sysbuild, ensuring the the MCUboot image has logging enabled and set to debug level, then when booting checking the message that will be printed on the board that mentions about slot sizes.
- overwrite only: images must be same size
- XIP: images must be same size

Reply:
Thanks your advice, I will adjust to fulfill overwrite only by tomorrow.

Reply:
Updated to let image 0 & image 1 same size.